samza-core/src/main/java/org/apache/samza/runtime/ContainerLaunchUtil.java

```diff
@@ -179,15 +181,26 @@ private static void run(
 
       if (containerRunnerException != null) {
         log.error("Container stopped with Exception. Exiting process now.", containerRunnerException);
-        System.exit(1);
+        exitProcess(1);
       }
     } catch (Throwable e) {
-      log.error("Container stopped with Exception. ", containerRunnerException);
+      log.error("Exiting the process due to {}. \nContainer runner exception: {}", e, containerRunnerException);
+      exitProcess(1);
     } finally {
       coordinatorStreamStore.close();
     }
   }
 
+  @VisibleForTesting
+  static CoordinatorStreamStore buildCoordinatorStreamStore(Config config, MetricsRegistryMap metricsRegistryMap) {
+    return new CoordinatorStreamStore(config, metricsRegistryMap);
+  }
+
+  @VisibleForTesting
```

Comment:
is this wrapping only for unit test purpose ?

Reply:
that is right

Reply:
I didn't want to go full on refactor this code. For now, I took a stab to enable unit testing my changes. 
Historically, this class has been pain to test and is missing coverage citing static & hard to mock code paths. We'd need a dedicated refactor and unit testing effort for this but taking the first step to add one.